.github/workflows/publish-gihub-release.yml

```diff
@@ -0,0 +1,28 @@
+---
+on:
+  push:
+    tags:
+    - v[0-9]+.[0-9]+.[0-9]+*
+
+jobs:
+  test:
+    uses: ./.github/workflows/.test.yml
+  push-docker:
+    uses: ./.github/workflows/.docker-push.yml
+    with:
+      is_pr: false
+    secrets:
+      dockerhub_username: "${{ secrets.DOCKERHUB_USERNAME }}"
+      dockerhub_token: "${{ secrets.DOCKERHUB_TOKEN }}"
+    needs: [test]
+  release:
+    runs-on: ubuntu-latest
+    container:
+      image: golang:1.18.0-bullseye
+    env:
+      GITHUB_TOKEN: "${{ secrets.GH_TOKEN }}"
+    steps:
+    - uses: actions/checkout@v3
```

Comment:
```suggestion
      - name: ci/checkout-repo
        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
```
Better use the pinned hash version for security hardening

Reply:
Great point, thanks!

Reply:
Also pinned all other reference to Actions